### PR TITLE
[CI] Use mesheryctl to start adapters in CI

### LINKED ISSUE #5360: Enhance GitHub workflow meshery-adapter tests to deploy adapters using mesheryctl.
#### Current Behavior
Workflows are using Kubernetes manifests to deploy Meshery adapters.

#### Desired Behavior
Edit the workflow in https://github.com/meshery/meshery/blob/master/.github/workflows/testadapters.yaml to use mesheryctl when deploying adapters.

#### Mockups

---
#### Contributor [Guides](https://docs.meshery.io/project/contributing) and Resources
- 🛠 [Meshery Build & Release Strategy](https://docs.meshery.io/project/build-and-release)
- 📚 [Instructions for contributing to documentation](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#documentation-contribution-flow)
   - Meshery documentation [site](https://docs.meshery.io/) and [source](https://github.com/meshery/meshery/tree/master/docs)
- 🎨 Wireframes and designs for Meshery UI in [Figma](https://www.figma.com/file/SMP3zxOjZztdOLtgN4dS2W/Meshery-UI)
- 🙋🏾🙋🏼 Questions: [Layer5 Discussion Forum](https://discuss.layer5.io) and [Layer5 Community Slack](http://slack.layer5.io)

### PR DESCRIPTION
**Pull request by gr455**
**Description**
Meshery adapters CI currently uses Kubernetes manifests to deploy meshery's service mesh adapters. This PR changes that to use mesheryctl to deploy the adapters by adding them to the meshery config.

This PR fixes #5360

Test run of Istio: https://github.com/gr455/meshery-istio/runs/5960526882?check_suite_focus=true

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

File: .github/workflows/testadapters.yaml [hunk 1]
```diff
@@ -82,23 +82,17 @@ jobs:
               echo '{ "meshery-provider": "None", "token": "" }'  > ~/.meshery/auth.json
             fi
             cat ~/.meshery/auth.json
-        
+
+      # Add adapter to meshconfig which will be picked up by mesheryctl to start the adapter
+      - name: Add adapter to meshconfig
+        run: |
+            yq -e -i '.contexts.local.components += ["meshery-${{ inputs.adapter_name }}"]' ~/.meshery/config.yaml
```
Comment (leecalcote) [inline .github/workflows/testadapters.yaml:89]: Perhaps, a more straight-forward approach is to create the meshconfig with specific adapters included (upon creation, like so `mesheryctl system context create -a <adapter-name>`).

Comment (gr455) [reply .github/workflows/testadapters.yaml:89]: Could not get this to work. 
```
Error: accepts 1 arg(s), received 0
```

Alternative is to create another context and add the adapter to it and switch contexts

Comment (leecalcote) [reply .github/workflows/testadapters.yaml:89]: @gr455  - - 


```
$ mesheryctl system context create test -a meshery-istio
Added `test` context

$ mesheryctl system context view test

Current Context: test

endpoint: http://localhost:9081
token: Default
token-location: auth.json
platform: kubernetes
components:
- meshery-istio
channel: stable
version: latest
```

**Review by leecalcote — APPROVED** [review]
Thank you, @gr455.  Will you demo this in this week's Build and Release meeting and speak to the status whether the `meshery-smp-action` and the e2e adapter tests are using this approach (hopefully, by meeting time on Thursday), they will be. // @Revolyssup @gyohuangxin @hershd23

**Comment by gr455** [comment]
> Thank you, @gr455. Will you demo this in this week's Build and Release meeting and speak to the status whether the `meshery-smp-action` and the e2e adapter tests are using this approach (hopefully, by meeting time on Thursday), they will be. // @Revolyssup @gyohuangxin @hershd23

Sure

**Comment by Revolyssup** [comment]
@leecalcote  @gr455  Wait I just saw this PR. I was using k8s manifest with image name of the adapter image we build during the workflow. Using mesheryctl we will always be running the stable adapter and not the one with the image build during the workflow.

**Comment by Revolyssup** [comment]
@leecalcote  @gr455  Unless mesheryctl gives us a way to override image name in the deployment.yaml of adapter, we cannot use mesheryctl for this specific case.

**Comment by gr455** [comment]
> @leecalcote  @gr455  Wait I just saw this PR. I was using k8s manifest with image name of the adapter image we build during the workflow. Using mesheryctl we will always be running the stable adapter and not the one with the image build during the workflow. This is an invalid test. Please revert this PR

Oh yes, obviously. This didn't even cross my mind.
Let's revert this then

**Comment by leecalcote** [comment]
Please give this more thought. Reverting isn't the right way forward.

**Comment by Revolyssup** [comment]
> Please give this more thought. Reverting isn't the right way forward.

As of now, what is being done here is not something that mesheryctl can do. We can create an issue in mesheryctl to have a feature to dynamically override image name. I asked to revert because this messed the core workflow. After this PR, all e2e tests across all adapters become obsolete.

**Pull request merged**